tests/run.sh

```diff
@@ -17,10 +17,14 @@ if [ "$CI" = true ]; then
     set -x
 fi
 
-source_dir="$PWD"
+source_dir="$(dirname "$(realpath "$(type -p $0)")")"/..
 build_dir="$CMAKE_BINARY_DIR"
 if [ -z "$build_dir" ]; then
-    build_dir="$PWD/build"
+    if [ -d "$source_dir/build" ]; then
+        build_dir="$source_dir/build"
+    else
+        build_dir="$PWD"
+    fi
```

Comment:
No, it wasn't supposed to be there. I still need this to run the tests using my setup (and the sleep 0.5), I accidentally pushed it again.